lib/rust/ensogl/component/scroll-area/src/lib.rs

```diff
@@ -278,6 +282,8 @@ impl ScrollArea {
 
             frp.source.scroll_position_x <+ model.h_scrollbar.thumb_position.map(|x| -x);
             frp.source.scroll_position_y <+ model.v_scrollbar.thumb_position;
+            frp.source.scroll_position_target_x <+ model.h_scrollbar.thumb_position_target.map(|x| -x);
+            frp.source.scroll_position_target_y <+ model.v_scrollbar.thumb_position_target;
```

Comment:
Are you sure you've covered all cases? What if scroll position was set by `scroll_to_x` or `scroll_to_y`?

Reply:
I'm pretty sure. `scroll_to_x` leads to `scrollbar.scroll_to`, which in turn sets the `thumb_position_target` here:
```rust
unbounded_target_position <- any(&frp.scroll_to,&frp.jump_to);
thumb_position.target     <+ all_with3(&unbounded_target_position,&frp.set_thumb_size,
                &frp.set_max,|target,&size,&max| target.min(max-size).max(0.0));
thumb_position.skip       <+ frp.jump_to.constant(());
frp.source.thumb_position_target <+ thumb_position.target;
frp.source.thumb_position <+ thumb_position.value;
```